control-plane/grpc/src/pool/traits.rs

```diff
@@ -0,0 +1,295 @@
+                };
+                Some(PoolSpec {
+                    node: pool_spec.clone().node_id.into(),
+                    id: pool_spec.clone().pool_id.into(),
+                    disks: pool_spec.disks.iter().map(|i| i.into()).collect(),
+                    status: pool_spec_status,
+                    labels: match pool_spec.labels {
+                        Some(labels) => Some(labels.value),
+                        None => None,
+                    },
+                    sequencer: Default::default(),
+                    operation: None,
+                })
+            }
+        };
+        Pool::try_new(pool_spec, pool_state).unwrap()
+    }
+}
+
+impl From<Pool> for pool_grpc::Pool {
+    fn from(pool: Pool) -> Self {
+        let pool_definition = match pool.spec() {
+            None => None,
+            Some(pool_spec) => Some(pool_grpc::PoolDefinition {
+                spec: Some(pool_grpc::PoolSpec {
+                    node_id: pool_spec.node.to_string(),
+                    pool_id: pool_spec.id.to_string(),
+                    disks: pool_spec.disks.iter().map(|i| i.to_string()).collect(),
+                    labels: pool_spec
+                        .labels
+                        .map(|labels| crate::common::StringMapValue { value: labels }),
+                }),
+                metadata: Some(pool_grpc::Metadata {
+                    uuid: None,
+                    status: pool_spec.status.into(),
+                }),
+            }),
+        };
+        let pool_state = match pool.state() {
+            None => None,
+            Some(pool_state) => Some(pool_grpc::PoolState {
+                node_id: pool_state.node.to_string(),
+                pool_id: pool_state.id.to_string(),
+                disks_uri: pool_state.disks.iter().map(|i| i.to_string()).collect(),
+                status: pool_state.status.into(),
+                capacity: pool_state.capacity,
+                used: pool_state.used,
+            }),
+        };
+        pool_grpc::Pool {
+            definition: pool_definition,
+            state: pool_state,
+        }
+    }
+}
+
+impl From<pool_grpc::Pools> for Pools {
+    fn from(pools: pool_grpc::Pools) -> Self {
+        Pools(pools.pools.iter().map(|pool| pool.clone().into()).collect())
+    }
+}
+
+impl From<Pools> for pool_grpc::Pools {
+    fn from(pools: Pools) -> Self {
+        pool_grpc::Pools {
+            pools: pools
+                .into_inner()
+                .iter()
+                .map(|pool| pool.clone().into())
+                .collect(),
+        }
+    }
+}
+
+impl From<i32> for crate::common::SpecStatus {
+    fn from(i: i32) -> Self {
+        match i {
+            0 => crate::common::SpecStatus::Creating,
+            1 => crate::common::SpecStatus::Created,
+            2 => crate::common::SpecStatus::Deleting,
+            3 => crate::common::SpecStatus::Deleted,
+            _ => crate::common::SpecStatus::Creating,
```

Comment:
This doesn't seem right? Should we have an unknown? or use a TryFrom ?
We can probably avoid this altogether if we use the grpc or newtype